atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClient.java

```diff
@@ -77,20 +107,67 @@ public void batch_mutate(Map<ByteBuffer, Map<String, List<Mutation>>> mutation_m
             super.batch_mutate(mutation_map, consistency_level);
             return null;
         });
+        try {
+            recordBytesWritten(getApproximateWriteByteCount(mutation_map));
+        } catch (Exception e) {
+            log.warn("Encountered an exception when recording write metrics for batch_mutate.", e);
+        }
+    }
+
+    private int getApproximateWriteByteCount(Map<ByteBuffer, Map<String, List<Mutation>>> mutation_map) {
+        int approxBytesForKeys = mutation_map.keySet().stream().mapToInt(e -> e.array().length).sum();
+        int approxBytesForValues = mutation_map.values().stream()
+                .mapToInt(singleMap -> {
+                    int approximateBytesInStrings = singleMap.keySet().stream().mapToInt(String::length).sum();
+                    int approximateBytesInMutations = singleMap.values().stream()
+                            .mapToInt(listOfMutations -> listOfMutations.stream()
+                                    .mapToInt(mutation -> SerializationUtils.serialize(mutation).length)
+                                    .sum())
+                            .sum();
+                    return approximateBytesInStrings + approximateBytesInMutations;
+                })
+                .sum();
+        return approxBytesForKeys + approxBytesForValues;
     }
 
     @Override
     public CqlResult execute_cql3_query(ByteBuffer query, Compression compression, ConsistencyLevel consistency)
             throws InvalidRequestException, UnavailableException, TimedOutException, SchemaDisagreementException,
             TException {
-        return checkLimitAndCall(() -> super.execute_cql3_query(query, compression, consistency));
+        CqlResult cqlResult = checkLimitAndCall(() -> super.execute_cql3_query(query, compression, consistency));
+        try {
+            recordBytesRead(SerializationUtils.serialize(cqlResult).length);
+        } catch (Exception e) {
+            log.warn("Encountered an exception when recording read metrics for execute_cql3_query.", e);
+        }
+        return cqlResult;
     }
 
     @Override
     public List<KeySlice> get_range_slices(ColumnParent column_parent, SlicePredicate predicate, KeyRange range,
             ConsistencyLevel consistency_level)
             throws InvalidRequestException, UnavailableException, TimedOutException, TException {
-        return checkLimitAndCall(() -> super.get_range_slices(column_parent, predicate, range, consistency_level));
+        List<KeySlice> result = checkLimitAndCall(
+                () -> super.get_range_slices(column_parent, predicate, range, consistency_level));
+        int approximateBytesRead = result.stream()
+                .mapToInt(keySlice -> SerializationUtils.serialize(keySlice).length)
+                .sum();
+        try {
+            recordBytesRead(approximateBytesRead);
+        } catch (Exception e) {
```

Comment:
I would really hope that recording metrics shouldn't throw an Exception for any reason.  Why do we need this?